client/tests/integration/upgrade.rs

```diff
@@ -33,17 +36,66 @@ fn validator_upgrade_should_work() -> Result<()> {
         .submit_transaction_blocking(&transfer_rose_tx)
         .expect_err("Should fail");
 
-    // Upgrade Validator
+    upgrade_validator(
+        &client,
+        "tests/integration/smartcontracts/validator_with_admin",
+    )?;
+
+    // Check that admin can transfer alice's rose now
+    // Creating new transaction instead of cloning, because we need to update it's creation time
+    let transfer_rose_tx = TransactionBuilder::new(admin_id)
+        .with_instructions([transfer_alice_rose])
+        .sign(admin_keypair)?;
+    client
+        .submit_transaction_blocking(&transfer_rose_tx)
+        .expect("Should succeed");
+
+    Ok(())
+}
+
+#[test]
+fn validator_upgrade_should_update_tokens() -> Result<()> {
+    let (_rt, _peer, client) = <PeerBuilder>::new().with_port(10_815).start_with_runtime();
+    wait_for_genesis_committed(&vec![client.clone()], 0);
+
+    // Check that `can_unregister_domain` exists
+    let can_unregister_domain_token_id: PermissionTokenId = "can_unregister_domain".parse()?;
+    let definitions = client.request(FindAllPermissionTokenDefinitions)?;
+    assert!(definitions
+        .into_iter()
+        .any(|definition| definition.id() == &can_unregister_domain_token_id));
+
+    upgrade_validator(
+        &client,
+        "tests/integration/smartcontracts/validator_with_custom_token",
+    )?;
+
+    // Check that `can_unregister_domain` doesn't exist
+    let definitions = client.request(FindAllPermissionTokenDefinitions)?;
+    assert!(!definitions
+        .iter()
+        .any(|definition| definition.id() == &can_unregister_domain_token_id));
+
+    // Check that `can_control_domain_lives` exists
+    let can_control_domain_lives: PermissionTokenId = "can_control_domain_lives".parse()?;
+    assert!(!definitions
+        .iter()
+        .any(|definition| definition.id() == &can_control_domain_lives));
```

Comment:
```suggestion
    // Check that `can_control_domain_lives` exists
    let can_control_domain_lives: PermissionTokenId = "can_control_domain_lives".parse()?;
    assert!(definitions
        .iter()
        .any(|definition| definition.id() == &can_control_domain_lives));
```

shouldn't this be like this? I noticed it fails in my PR

Reply:
Right. Copy-paste mistake. I wonder why it doesn't fail for me

Reply:
It's because of typo